pkg/controller/corecerts/corecerts_controller.go

```diff
@@ -0,0 +1,228 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package corecerts
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/golang/glog"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	corev1informers "k8s.io/client-go/informers/core/v1"
+	"k8s.io/client-go/kubernetes"
+	corev1listers "k8s.io/client-go/listers/core/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+
+	"io/ioutil"
+
+	clientset "github.com/xmudrii/etcdproxy-controller/pkg/client/clientset/versioned"
+)
+
+const (
+	controllerAgentName = "corecerts-controller"
+
+	epcNamespace = "kube-apiserver-storage"
+)
+
+type ControllerData struct {
+	EtcdCoreCACert string
+	EtcdCoreCert   string
+	EtcdCoreKey    string
+}
+
+// Controller is the controller implementation for EtcdStorage resources
+type Controller struct {
+	// kubeclientset is a standard kubernetes clientset
+	kubeclientset kubernetes.Interface
+	// sampleclientset is a clientset for our own API group
+	sampleclientset clientset.Interface
+
+	secretsLister corev1listers.SecretLister
+	secretsSynced cache.InformerSynced
+
+	workqueue workqueue.RateLimitingInterface
+
+	controllerData *ControllerData
```

Comment:
General design note.  You want your controller type at this level to have all the data it needs, so you don't want to pass filenames.  Instead you want to pass byte arrays that contain the content of the file.  That way your unit tests can stub the bytes in directly and you'll get "fail early" behavior in your command.